src/pg_query_summary_statement_type.c

```diff
@@ -0,0 +1,798 @@
+			break;
+		case T_AlterDatabaseStmt:
+			add_stmt_type_if_needed(summary, "AlterDatabaseStmt");
+			break;
+		case T_AlterDefaultPrivilegesStmt:
+			add_stmt_type_if_needed(summary, "AlterDefaultPrivilegesStmt");
+			break;
+		case T_AlterDomainStmt:
+			add_stmt_type_if_needed(summary, "AlterDomainStmt");
+			break;
+		case T_AlterEnumStmt:
+			add_stmt_type_if_needed(summary, "AlterEnumStmt");
+			break;
+		case T_AlterEventTrigStmt:
+			add_stmt_type_if_needed(summary, "AlterEventTrigStmt");
+			break;
+		case T_AlterExtensionContentsStmt:
+			add_stmt_type_if_needed(summary, "AlterExtensionContentsStmt");
+			break;
+		case T_AlterExtensionStmt:
+			add_stmt_type_if_needed(summary, "AlterExtensionStmt");
+			break;
+		case T_AlterFdwStmt:
+			add_stmt_type_if_needed(summary, "AlterFdwStmt");
+			break;
+		case T_AlterForeignServerStmt:
+			add_stmt_type_if_needed(summary, "AlterForeignServerStmt");
+			break;
+		case T_AlterFunctionStmt:
+			add_stmt_type_if_needed(summary, "AlterFunctionStmt");
+			break;
+		case T_AlterObjectDependsStmt:
+			add_stmt_type_if_needed(summary, "AlterObjectDependsStmt");
+			break;
+		case T_AlterObjectSchemaStmt:
+			add_stmt_type_if_needed(summary, "AlterObjectSchemaStmt");
+			break;
+		case T_AlterOpFamilyStmt:
+			add_stmt_type_if_needed(summary, "AlterOpFamilyStmt");
+			break;
+		case T_AlterOperatorStmt:
+			add_stmt_type_if_needed(summary, "AlterOperatorStmt");
+			break;
+		case T_AlterOwnerStmt:
+			add_stmt_type_if_needed(summary, "AlterOwnerStmt");
+			break;
+		case T_AlterPolicyStmt:
+			add_stmt_type_if_needed(summary, "AlterPolicyStmt");
+			break;
+		case T_AlterPublicationStmt:
+			add_stmt_type_if_needed(summary, "AlterPublicationStmt");
+			break;
+		case T_AlterRoleSetStmt:
+			add_stmt_type_if_needed(summary, "AlterRoleSetStmt");
+			break;
+		case T_AlterRoleStmt:
+			add_stmt_type_if_needed(summary, "AlterRoleStmt");
+			break;
+		case T_AlterSeqStmt:
+			add_stmt_type_if_needed(summary, "AlterSeqStmt");
+			break;
+		case T_AlterStatsStmt:
+			add_stmt_type_if_needed(summary, "AlterStatsStmt");
+			break;
+		case T_AlterSubscriptionStmt:
+			add_stmt_type_if_needed(summary, "AlterSubscriptionStmt");
+			break;
+		case T_AlterSystemStmt:
+			add_stmt_type_if_needed(summary, "AlterSystemStmt");
+			break;
+		case T_AlterTableCmd:
+			add_stmt_type_if_needed(summary, "AlterTableCmd");
+			break;
+		case T_AlterTableMoveAllStmt:
+			add_stmt_type_if_needed(summary, "AlterTableMoveAllStmt");
+			break;
+		case T_AlterTableSpaceOptionsStmt:
+			add_stmt_type_if_needed(summary, "AlterTableSpaceOptionsStmt");
+			break;
+		case T_AlterTableStmt: {
+			add_stmt_type_if_needed(summary, "AlterTableStmt");
+			// Return early because raw_expression_tree_walker doesn't handle this.
+			return false;
+		}
+		case T_AlterTSConfigurationStmt:
+			// Capitalization is deliberately different to match Rust implementation.
+			add_stmt_type_if_needed(summary, "AlterTsconfigurationStmt");
```

Comment:
I think we should intentionally break from the Rust implementation here - this was not an intentional choice (to my knowledge), and it'd be good to exactly match the C struct names.